apps/google-analytics-4/frontend/src/components/main-app/analytics-app/AnalyticsApp.tsx

```diff
@@ -41,30 +57,20 @@ const AnalyticsApp = () => {
     }
 
     fetchData();
-  }, []);
+  }, [api, reportRequestParams]);
 
   useEffect(() => {
-    const sliceByDateRange = (dateRange: DateRangeType): RunReportResponse => {
-      let newRows: Row[] = [];
-      switch (dateRange) {
-        case DateRange.LastDay:
-          newRows = runReportResponse.rows.slice(0, 2);
-          break;
-        case DateRange.LastWeek:
-          newRows = runReportResponse.rows.slice(0, 7);
-          break;
-        case DateRange.LastMonth:
-          newRows = runReportResponse.rows.slice(0, 28);
-          break;
-      }
-      return { ...runReportResponse, rows: [...newRows] };
-    };
     if (runReportResponse.rowCount) {
-      const newData = sliceByDateRange(dateRange);
-      setPageViewData(newData);
+      setStartEndDates(getRangeDates(dateRange));
+      setRunReportResponse(runReportResponse);
+      setLoading(false);
```

Comment:
I don't think there will be a bug here per-se atm but you have two different useEffects setting the loading state to false. So if one useEffect completes, then the component will render below. 
1. Either you need two different loading states or marriage the useEffects together to settle the loading state

Reply:
ah yes I believe this is remnant of the previous logic we had slicing the dates manually instead of letting the API handle it. Thanks for pointing this out, I am going to remove one of these loadings states !